src/reflect/scala/reflect/internal/transform/UnCurry.scala

```diff
@@ -37,87 +36,80 @@ trait UnCurry {
    *  we instead get
    *    Vector(tab, tb, tab)
    *  I think that difference is not the product of sentience but of randomness.
-   *  Let us figure out why it is and then change this method.
+   *  The reason for this are nested RefinedTypes that are flattened by tp.normalize.
+   *
+   *  And normalizing AnyRef leads to another single test failure: run/t2503.scala.
    */
-  private def expandAlias(tp: Type): Type = if (!tp.isHigherKinded) tp.normalize else tp
+  private def expandAlias(tp: Type): Type =
+    if (!tp.isHigherKinded && tp.typeSymbolDirect != AnyRefClass) tp.normalize else tp
```

Comment:
Meh i had a longer comment here that got lost...

Let me try again. I'm a little worried about the AnyRef special case.

I digged a bit and what breaks is `rebindSuper` in mixin failing to find the super call target.

```
trait SB[A] extends Buffer[A] {
  abstract override def update(n: Int, newelem: A): Unit = synchronized {
    super.update(n, newelem)
  }
}
new ArrayBuffer[Int] with SB[Int]
```

rebindSuper does
  - `acc.matchingSymbol(bcs.head, site)` where acc=mixin accessor for update, bcs.head = ArrayBuffer, site = ArrayBuffer[Int] with SB[Int]
  - the underlying problem is that `acc.tpe.asSeenFrom(site, acc.owner)` is `(idx: Int, elem: A): Unit` instead of `(idx: Int, elem: Int): Unit`
  - since the whole thing runs `exitingSpecialize` I tried various phases, and it seems the change is entering/exiting the fields phase. we get the right signature before fields. i have no idea why...

Reply:
This is just a long way of saying that this is pretty complex territory...

Reply:
I wish I could understand why `AnyRef` vs `Object` makes a difference.

Reply:
The weirdest thing is that we get the correct signature for `insertAll` but nor for `update` ...